scripts/gen_mapfiles.py

```diff
@@ -0,0 +1,47 @@
+"""
+    scripts/gen_mapfiles.py
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    Regenerate mapping files.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from importlib import import_module
+from pathlib import Path
+import re
+import sys
+
+top_src_dir = Path(__file__).parent.parent
+pygments_package = top_src_dir / 'pygments'
+sys.path.insert(0, pygments_package.parent.resolve())
+
+from pygments.util import docstring_headline
+
+for key in ['lexers', 'formatters']:
+    lines = []
+    for file in (pygments_package / key).glob('[!_]*.py'):
+        module_name = '.'.join(file.relative_to(pygments_package.parent).with_suffix('').parts)
+        print(module_name)
+        module = import_module(module_name)
+        for obj_name in module.__all__:
+            obj = getattr(module, obj_name)
+            desc = (module_name, obj.name, tuple(obj.aliases), tuple(obj.filenames))
+            if key == 'lexers':
+                desc += (tuple(obj.mimetypes),)
+            else: # key == 'formatters'
```

Comment:
I'd prefer a `elif key == 'formatters'` and explicit error handling otherwise. Yes it's stupid because there are only two types here, but if someone ever adds something to the list, this branch will cause trouble instead of failing early.

Also, shouldn't this be `if` in both cases, i.e.:
```python
if key == 'lexers':
  # add lexer specific data to desc
if key == 'formatters':
  # add formatter specific data to desc
```

(Also also, I wish we could require Python 3.10 here and use `match` ...)

Reply:
Done, but with `elif` since otherwise I don't have an `else` to raise an error.